plugins/zenoh-plugin-storage-manager/src/replication/core/aligner_query.rs

```diff
@@ -272,60 +269,74 @@ impl Replication {
         reply_to_query(query, reply, None).await;
     }
 
-    /// Replies to the [Query] with the [EventMetadata] and [Value] that were identified as missing.
+    /// Replies to the [Query] with the [EventMetadata] and [Value] identified as missing.
     ///
-    /// This method will fetch the [StoredData] from the Storage for each provided [EventMetadata],
-    /// making a distinct reply for each. The fact that multiple replies are sent to the same Query
-    /// is the reason why we need the consolidation to set to be `None` (⚠️).
-    pub(crate) async fn reply_events(&self, query: &Query, events_to_retrieve: Vec<EventMetadata>) {
-        for event_metadata in events_to_retrieve {
-            if event_metadata.action == SampleKind::Delete {
-                reply_to_query(query, AlignmentReply::Retrieval(event_metadata), None).await;
-                continue;
-            }
+    /// Depending on the associated action, this method will fetch the [Value] either from the
+    /// Storage or from the wildcard updates.
+    pub(crate) async fn reply_event_retrieval(
+        &self,
+        query: &Query,
+        event_to_retrieve: EventMetadata,
+    ) {
+        let value = match &event_to_retrieve.action {
+            // For a Delete or WildcardDelete there is no associated `Value`.
+            Action::Delete | Action::WildcardDelete(_) => None,
+            // For a Put we need to retrieve the `Value` in the Storage.
+            Action::Put => {
+                let stored_data = {
+                    let mut storage = self.storage_service.storage.lock().await;
+                    match storage
+                        .get(event_to_retrieve.stripped_key.clone(), "")
+                        .await
+                    {
+                        Ok(stored_data) => stored_data,
+                        Err(e) => {
+                            tracing::error!(
+                                "Failed to retrieve data associated to key < {:?} >: {e:?}",
+                                event_to_retrieve.key_expr()
```

Comment:
The log currently looks something like
```
2024-10-08T14:58:30.624185Z ERROR tokio-runtime-worker ThreadId(02) zenoh_plugin_storage_manager::replication::core::aligner_query: Failed to retrieve data associated to key < Some("wildcard/delete/a/1") >: "Key Some(\"wildcard/delete/a/1\") is not present"
```

Maybe we should unwrap if possible, for a cleaner error message.